src/common/utils.py

```diff
@@ -1,27 +1,35 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-import sys
+from . import constants as CONSTANTS
 import json
 import copy
 import time
-from . import constants as CONSTANTS
-from . import debugger_communication_client
-from applicationinsights import TelemetryClient
+import sys
 
 previous_state = {}
 
 
-def show(state, debug_mode=False):
+def create_message(state, device_name):
+    state_copy = dict(state)
+    state_ext = {
+        "device_name": device_name,
+    }
+    state_copy.update(state_ext)
+
+    message = {"type": "state", "data": json.dumps(state_copy)}
+    return message
+
+
+def send_to_simulator(state, device_name):
     global previous_state
+
+    message = create_message(state, device_name)
+
     if state != previous_state:
```

Comment:
Thanks for addressing the comment!
But now this part might not work as expected, right?
I would take the state updating logic and put it in this method (or in a function and call that function here).

Reply:
Thanks! I just abstracted out the state updating and made the process two function calls :).

Reply:
Awesome!
One more thing, now `send_to_simulator` and `debug_show` mutate the input parameters. It would be easier for a developer if they can use these functions without worrying about any unintended consequences.

Reply:
ok, tried to fix it in the last commit :)

Reply:
Almost perfect!
Now this check will always return true `if state != previous_state:`. Perhaps use `state_copy` instead?

Reply:
oops! fixed

Reply:
_&ast;nit&ast;_
I would change the name to `updated_state` from `state_copy`. Up to you!

Reply:
fixed!

Reply:
LGTM!
Thanks for your patience :D
![giphy](https://user-images.githubusercontent.com/12337821/74182699-6663a380-4bf8-11ea-95d7-c24eeab69e7f.gif)